tools/cli/admin_task_list_commands.go

```diff
@@ -200,25 +208,78 @@ func AdminUpdateTaskListPartitionConfig(c *cli.Context) error {
 	if err != nil {
 		return commoncli.Problem("Error in creating context:", err)
 	}
+	cfg := &types.TaskListPartitionConfig{
+		ReadPartitions:  createPartitions(numReadPartitions),
+		WritePartitions: createPartitions(numWritePartitions),
+	}
+	tl := &types.TaskList{Name: taskList, Kind: types.TaskListKindNormal.Ptr()}
+	if !force {
+		err = validateChange(ctx, frontendClient, domain, tl, taskListType, cfg)
+		if err != nil {
+			return commoncli.Problem("Potentially unsafe operation. Specify '--force' to proceed anyway: ", err)
+		}
+	}
 	_, err = adminClient.UpdateTaskListPartitionConfig(ctx, &types.UpdateTaskListPartitionConfigRequest{
-		Domain:       domain,
-		TaskList:     &types.TaskList{Name: taskList, Kind: types.TaskListKindNormal.Ptr()},
-		TaskListType: taskListType,
-		PartitionConfig: &types.TaskListPartitionConfig{
-			ReadPartitions:  createPartitions(numReadPartitions),
-			WritePartitions: createPartitions(numWritePartitions),
-		},
+		Domain:          domain,
+		TaskList:        tl,
+		TaskListType:    taskListType,
+		PartitionConfig: cfg,
 	})
 	if err != nil {
 		return commoncli.Problem("Operation UpdateTaskListPartitionConfig failed.", err)
 	}
 	return nil
 }
 
+func validateChange(ctx context.Context, client frontend.Client, domain string, tl *types.TaskList, tlt *types.TaskListType, newCfg *types.TaskListPartitionConfig) error {
+	description, err := client.DescribeTaskList(ctx, &types.DescribeTaskListRequest{
+		Domain:       domain,
+		TaskList:     tl,
+		TaskListType: tlt,
+	})
+	if err != nil {
+		return fmt.Errorf("DescribeTaskList failed: %w", err)
+	}
+	if len(description.Pollers) == 0 {
+		return fmt.Errorf("'%s' has no pollers of type '%s'", tl.Name, tlt.String())
```

Comment:
I wonder if this check should be just warning or move it to the end. Operator seeing this error might just rerun with `-f` without knowing about the dangerous stuff checked below.

Reply:
Good call out, I'm going to move this check to the end